### PR TITLE
58733_category_management

### PR DESCRIPTION
**Pull request by ahammad-reyad**
category management implemented

### CODE REVIEW

File: app/controllers/admin/categories_controller.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  def index
+    @categories = Category.paginate page: params[:page]
+  end
+
+  def new
+    @category = Category.new
+  end
+
+  def show
```
Comment (MuhammadTamzid) [inline app/controllers/admin/categories_controller.rb]: no implementation 
remove for this patch

File: app/controllers/admin/categories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    if @category.save
+      flash[:success] = t :category_created
+      redirect_to admin_categories_path
+    else
+      render :new
+    end
+  end
+
+  def edit
+    #@category = Category.find params[:id]
```
Comment (MuhammadTamzid) [inline app/controllers/admin/categories_controller.rb]: remove

File: app/controllers/admin/categories_controller.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  end
+
+  def update
+    #@category = Category.find params[:id]
```
Comment (MuhammadTamzid) [inline app/controllers/admin/categories_controller.rb]: remove

File: app/controllers/admin/categories_controller.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    if @category.update_attributes(category_params)
```
Comment (MuhammadTamzid) [inline app/controllers/admin/categories_controller.rb]: remove ()

File: config/routes.rb [hunk 5]
```diff
@@ -7,9 +7,11 @@
   post   "login"  => "sessions#create"
   delete "logout" => "sessions#destroy"
   resources :users
+  resources :categories
```
Comment (MuhammadTamzid) [inline config/routes.rb]: no implementation found in this patch

File: db/seeds.rb [hunk 6]
```diff
@@ -13,3 +13,15 @@
     password: password,
     password_confirmation: password)
 end
+
+users = User.all
+user  = users.first
+following = users[2..50]
+followers = users[3..40]
+following.each { |followed| user.follow(followed) }
```
Comment (MuhammadTamzid) [inline db/seeds.rb]: ``
following.each {|followed| user.follow(followed)}
``

File: db/seeds.rb [hunk 7]
```diff
@@ -13,3 +13,15 @@
...
+followers.each { |follower| follower.follow(user) }
```
Comment (MuhammadTamzid) [inline db/seeds.rb]: ``
followers.each {|follower| follower.follow(user)}
``

File: db/seeds.rb [hunk 8]
```diff
@@ -13,3 +13,15 @@
...
+
+100.times do |n|
+  category_name  = "Category-#{n+1}"
+  Category.create!(category_name: category_name)
```
Comment (MuhammadTamzid) [inline db/seeds.rb]: remove ()

File: db/seeds.rb [hunk 9]
```diff
@@ -13,3 +13,15 @@
...
+users = User.all
+user  = users.first
+following = users[2..50]
+followers = users[3..40]
+following.each {|followed| user.follow(followed)}
+followers.each {|follower| follower.follow(user)}
+
+100.times do |n|
+  category_name  = "Category-#{n+1}"
+  Category.create!(category_name: category_name)
```
Comment (MuhammadTamzid) [inline db/seeds.rb]: remove ()

File: app/controllers/sessions_controller.rb [hunk 10]
```diff
@@ -8,7 +8,7 @@ def create
       log_in user
       remember user
       params[:session][:remember_me] == "1" ? remember(user) : forget(user)
-      user.is_admin? redirect_to(admin_users_path) : redirect_back_or(user)
+      user.is_admin ? redirect_to(admin_users_path) : redirect_back_or(user)
```
Comment (dieunb) [inline app/controllers/sessions_controller.rb]: `user.is_admin? ? redirect_to(admin_users_path) : redirect_back_or(user)`

File: app/views/admin/categories/_category.html.erb [hunk 11]
```diff
@@ -0,0 +1,7 @@
+<li>
+  <%= link_to category.category_name, edit_admin_category_path(category) %>
+  <% if current_user.is_admin? %>
+    | <%= link_to t(:delete), admin_category_path(category), method: :delete,
+    data: {confirm: t(:delete_confirm)} %>
```
Comment (dieunb) [inline app/views/admin/categories/_category.html.erb]: indent

**Pull request merged**